python/ray/autoscaler/updater.py

```diff
@@ -133,7 +137,10 @@ def do_update(self):
 
         node_tags = self.provider.node_tags(self.node_id)
         logger.debug("Node tags: {}".format(str(node_tags)))
-        if node_tags.get(TAG_RAY_RUNTIME_CONFIG) == self.runtime_hash:
+        if node_tags.get(
+                TAG_RAY_RUNTIME_CONFIG) == self.runtime_hash and node_tags.get(
+                    TAG_RAY_FILE_MOUNTS_CONTENTS
+                ) == self.file_mounts_contents_hash:
```

Comment:
Can we split this to two different cases (continuous vs continuous, it will also clean the if in the else)? or at least add a comment saying that we do not update self.file_mounts_contents_hash in noncontinuous file sync mode and thus it will always be equal to node_tags[TAG_RAY_FILE_MOUNTS_CONTENTS]. Can we not set these tags/check them if not in continuous sync mode?

Reply:
The contents of the file_mounts were hashed previously when continuous mode didn't even exist. I think its necessary so that that the autoscaler can detect changes to the file_mounts contents whenever the user runs `ray up`.

I prefer to add a comment vs separating out the two cases since the two cases are very similar except for 1 code path.

Reply:
I see. Makes sense.